### PR TITLE
route out some python 3.4 errors

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by Tarrasch** [comment]
Can you motivate this a bit more? I'm afraid I don't understand what this does and why.

**Comment by erikbern** [comment]
also confused :)

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 language: python
+python: 3.4
```
Comment (Tarrasch) [inline .travis.yml]: What does this do?

**Comment by Tarrasch** [comment]
@graingert: If you intend to work on a PR continuously. It's a nice practice to say "WIP" (work in progress) so we don't start to review it before the work is done. :)

File: .travis.yml (hunk 1 above)
Comment (graingert) [inline .travis.yml]: It means Travis creates a python v3.4 virtualenv

Comment (Tarrasch) [inline .travis.yml]: Is it needed if we use tox in the way that we do?

**Comment by graingert** [comment]
@Tarrasch it's no longer WIP....

**Comment by graingert** [comment]
@Tarrasch basically a few lint errors arn't being picked up by travis

File: luigi/contrib/hdfs/snakebite_client.py [hunk 2]
```diff
@@ -50,7 +50,9 @@ def __init__(self):
     def list_path(path):
         if isinstance(path, list) or isinstance(path, tuple):
             return path
-        if isinstance(path, str) or isinstance(path, unicode):
+        # TODO: Should this be:
+        # isinstance(path, (six.text_type, six.binary_type))?
+        if isinstance(path, six.string_types):
```
Comment (Tarrasch) [inline luigi/contrib/hdfs/snakebite_client.py]: Do you know the answer to the TODO @gpoulin ?

**Comment by Tarrasch** [comment]
Much cool. Anyone else wants to review? LGTM

**Pull request merged**